packages/@glimmer/runtime/test/partial-test.ts

```diff
@@ -324,6 +324,67 @@ QUnit.test('nested dynamic partial with dynamic content', () => {
   equalTokens(root, `Before <div>Testing wat are <div>Nested you doing?</div></div> After`);
 });
 
+QUnit.test('nested partial within a `{{#with}}` block', () => {
+  let template = compile(`{{#with 'Sophie' as |name|}}Before {{partial 'mantle'}} After{{/with}}`);
+
+  env.registerPartial('mantle', `[mantle {{name}} ({{age}}) {{partial 'outer'}}]`);
+  env.registerPartial('outer', `[outer {{name}} ({{age}}) {{partial 'inner'}}]`);
+  env.registerPartial('inner', `[inner {{name}} ({{age}})]`);
+
+  render(template, { age: 0 });
+
+  equalTokens(root, `Before [mantle Sophie (0) [outer Sophie (0) [inner Sophie (0)]]] After`);
+  rerender({ age: 0 }, { assertStable: true });
+  equalTokens(root, `Before [mantle Sophie (0) [outer Sophie (0) [inner Sophie (0)]]] After`);
+});
+
+QUnit.test('nested partials within nested `{{#with}}` blocks', () => {
+  let template = compile(`{{#with 'Sophie' as |person1|}}Hi {{person1}} (aged {{age}}), {{person2}}, {{person3}} and {{person4}}. {{partial 'person2-partial'}}{{/with}}`);
```

Comment:
Hm, I think I'll change this to

`Hi {{person1}}. {{#with 'Sophie' as |person1|}}Hi {{person1}}...`

so that we render something from the outer conext

Reply:
👍 sounds good

Reply:
updated